components/codeblock/codeblock.vue

```diff
@@ -0,0 +1,20 @@
+<template>
+  <pre>
+    <code
+      class="d-d-block d-bgc-white d-ws-pre-wrap d-p8 d-my4 d-ba d-bc-black-500 d-bar8 d-fs-100 d-lh-200 d-fc-red-400"
+    >{{ text }}</code>
+  </pre>
+</template>
+
+<script>
+export default {
+  name: 'DtCodeblock',
+
+  props: {
+    text: {
+      type: String,
+      required: true,
+    },
+  },
```

Comment:
Having this as a prop rather than a slot is totally fine for passing in user entered data, however it might be pretty awkward if you were using this for something like code examples on a documentation site. Imagine copy pasting your code example and then having to add a whole bunch of \n and such manually. Anyways not something we have to change now I don't think. Just thinking about future usage.

Reply:
Totally agree. Might be a case of YAGNI but one reason I went with the text prop instead of a slot was for if/when we add syntax highlighting. [The way we currently do it](https://github.com/dialpad/firespotter/blob/0ea598ca7a57b5580692bc391031f2b227c2822c/ubervoice/static/js/single/components/feed/rich_media/feed_rich_media_highlight.vue#L3-L10), we format the text with _highlight.js_ and then `v-html` it into the codeblock component. We might be able to find alternative ways of doing this with a slot though. What are your thoughts?

Reply:
That would be awesome. If any library is in play has some theming options, work with @bhawkins-dialpad to choose one that works best per his judgment, and identify a dark one to future-proof for dark mode.

Reply:
@bhawkins-dialpad  @francisrupert Looks like highlight.js has [a lot of good theming options for dark mode](https://highlightjs.org/static/demo/), I need to do some research into how we can conditionally choose a theme depending on if we're in dark mode or not.

@braddialpad good find with the Vue library! I saw in the readme for the one linked that it's been superseded by the new [native Highlight.js Vue plugin](https://github.com/highlightjs/vue-plugin). Do you have any objections to using that?

Reply:
> I saw in the readme for the one linked that it's been superseded by the new [native Highlight.js Vue plugin](https://github.com/highlightjs/vue-plugin). Do you have any objections to using that?

Oh yeah, definitely use that.

Reply:
Wow, so many options. Would you agree it probably doesn't matter what we choose now, as long as it's dead simple to pick another at any point? If so, don't consider this task blocked by that, and at another time the Dialtone Design team will explore what feels best with the Dialtone color palette. 

Personally, I've changed mine so many times but always come back to [Tomorrow night](https://highlightjs.org/static/demo/#Base16%20/%20Tomorrow%20Night).

Reply:
Yeah the highlighting is not even part of this story, it will be a future update.

Reply:
Good points. Just to confirm, is it ok to keep the interface as a text prop for now instead of a slot then?

Reply:
Yep